sources/common-dylan/byte-vector.dylan

```diff
@@ -92,7 +92,7 @@ define open method copy-bytes
      src :: <string>, src-start :: <integer>, n :: <integer>)
  => ()
   for (i :: <integer> from 0 below n)
-    dst[dst-start + i] := as(<integer>, src[src-start + i])
+    dst[dst-start + i] := src[src-start + i]
```

Comment:
Perhaps there shouldn't be methods on general `<vector>` at all. (We can leave this for now, but for the unicode branch I'll be seriously curtailing the use of `copy-bytes` where it's not appropriate.)